packages/client/wallets/aa/src/passkeys/services/LitService.ts

```diff
@@ -0,0 +1,204 @@
+import { logError, logInfo } from "@/services/logging";
+import { LitProtocolError, errorToJSON, isLocalhost } from "@/utils";
+import { POLYGON_CHAIN_ID, RELAY_API_KEY } from "@/utils/constants";
+import { LitAbility, LitActionResource } from "@lit-protocol/auth-helpers";
+import { ProviderType } from "@lit-protocol/constants";
+import { LitAuthClient, WebAuthnProvider } from "@lit-protocol/lit-auth-client";
+import { LitNodeClient, decryptToString, encryptString } from "@lit-protocol/lit-node-client";
+import { PKPEthersWallet } from "@lit-protocol/pkp-ethers";
+import { AuthSig } from "@lit-protocol/types";
+import { SiweMessage } from "siwe";
+
+const chain = "polygon";
+
+export class LitService {
+    private litNodeClient: LitNodeClient | undefined;
+    private litAuthClient: LitAuthClient | undefined;
+
+    async connect() {
+        this.litAuthClient = new LitAuthClient({
+            litRelayConfig: {
+                relayApiKey: RELAY_API_KEY,
+            },
+        });
+        this.litAuthClient.initProvider<WebAuthnProvider>(ProviderType.WebAuthn);
+        this.litNodeClient = new LitNodeClient({
+            litNetwork: isLocalhost() ? "manzano" : "habanero",
+        });
+        await this.litNodeClient.connect();
+    }
+
+    async registerWithWebAuthn(identifier: string) {
+        try {
+            if (this.litAuthClient == null) {
+                await this.connect();
+            }
+            const provider = this.litAuthClient!.getProvider(ProviderType.WebAuthn) as WebAuthnProvider;
+            // Register new WebAuthn credential
+            const options = await provider!.register(identifier);
+            // Verify registration and mint PKP through relay server
+            const txHash = await provider!.verifyAndMintPKPThroughRelayer(options);
+            const response = await provider.relay.pollRequestUntilTerminalState(txHash);
+
+            if (response.status !== "Succeeded") {
+                throw new Error("Failed to register with WebAuthn");
+            }
+
+            logInfo("[LIT_REGISTER_WEBAUTHN] - FINISH", { identifier });
+            return {
+                pkpEthAddress: response.pkpEthAddress,
+                pkpPublicKey: response.pkpPublicKey,
+            };
+        } catch (error: any) {
+            logError("[LIT_REGISTER_WEBAUTHN] - ERROR_REGISTER_WEBAUTHN", { error: errorToJSON(error), identifier });
+            throw new LitProtocolError(`Error signing up [${error?.name ?? ""}]`);
+        }
+    }
+    async encrypt(
+        messageToEncrypt: string,
+        pkpPublicKey: string,
+        pkpEthAddress: string,
+        capacityDelegationAuthSig: AuthSig
+    ) {
```

Comment:
because js doesn't have argument labels, it can be easy to fuck up the params when there are consecutive args of the same type, so I'll typically convert this into an object to bring back those labels:
```
type EncryptInput = {
    message: string;
    pkpPublicKey: string;
    pkpEthAddress: string;
    capacityDelegationAuthSig: AuthSig;
}

encrypt({ messageToEncrypt, pkpPublicKey, pkpEthAddress, capacityDelegationAuthSig }: EncryptInput) { ... }
```
so that the call-site is:
```
encrypt({ message: "...", pkpPublicKey: ..., pkpEthAddress: ...